### ISSUE TITLE
db:migrate:with_data not picking up schema changes

### ISSUE THREAD

**Issue by jasonivers** [ilyakatz/data-migrate#158]
I have two migrations, one a schema update to add a column to a table, the other a data migration to set the right data for existing records for that field.  The schema migration has an id/timestamp/whatever of 20201111204508, and the data migration has an id of 20201111204520.  The schema migration logs out to the screen first (as it should), but the data migration fails because it doesn't see the new column in the table (or attribute on the model).  If I then, after all migrations finish, just run rails data:migrate, the same migration runs just fine.

Do I need to call reset_column_information on the model in the data migration before updating the new field, or is it something else?

**Comment by saiidalhalawi** [comment]
I've faced same issue, and this seems to be caused by below part.

https://github.com/ilyakatz/data-migrate/blob/master/tasks/databases.rake#L33-L38

As far as I can see, `schema_migrations` versions don't be executed if both `schema_migrations` and  `data_migrations` are picked.

**Comment by pjmartorell** [comment]
@saiidalhalawi I don't think this is the issue, if there are pending `schema_migrations` and `data_migrations` they are [sorted](https://github.com/ilyakatz/data-migrate/blob/2dd90c495b4d57e4dc700e3f9be149c7e2b93b57/tasks/databases.rake#L38) and run by order of timestamp, no issues there. The issue is that, when the model is loaded, its schema is being cached and new created columns are not visible by Rails except you call `reset_column_information`.

**Comment by sully7** [comment]
Is this related to #201 by chance?

**Comment by ovalenzuela19** [comment]
splitting the schema and data migration commands fixes the issue. run `db:migrate` first, then `db:migrate:with_data` after
